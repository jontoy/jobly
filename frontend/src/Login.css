.Login {
    text-align: left;
}

.Login button {
    text-align: center;
}