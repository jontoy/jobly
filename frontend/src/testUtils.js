import React from "react";
import UserContext from "./userContext";

const demoUser = {
  username: "test",
  first_name: "test",
  last_name: "mctest",
  email: "test@test.net",
  photo_url: null,
  jobs:[]
};

const UserProvider = ({ children, currentUser = demoUser, appliedIds = [] }) => (
  <UserContext.Provider value={{currentUser, appliedIds}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
