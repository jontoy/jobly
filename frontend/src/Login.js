import React, {useState, useContext} from 'react';
import JoblyApi from './JoblyApi';
import {useHistory} from 'react-router-dom';
import UserContext from './userContext';
import './Login.css';
import Alert from './Alert';

const Login = () => {
    const {setToken} = useContext(UserContext);
    const history = useHistory();
    const [isNewUser, setIsNewUser] = useState(false);
    const BASE_STATE = {username:'', password:'', first_name:'', last_name:'', email:'', errors:[]};
    const [formData, setFormData] = useState(BASE_STATE);
    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(data => ({
            ...data,
            errors: [],
            [name]: value

        }));
    }
    const handleSubmit = async (e) => {
        e.preventDefault();
        let token;
        try{
            const {errors, ...data} = formData;
            if(isNewUser){
                token = await JoblyApi.signup(data);
            } else{
                token = await JoblyApi.login(data);
            }
            setToken(token);
            history.push('/');
        } catch (errors){
            setFormData(data => ({
                ...data,
                errors
            }));
        }

    }
    return (
        <div className="Login container">
            <div className="card">
                <div className="card-header">
                    <ul className="nav nav-tabs card-header-tabs ml-auto">
                        <li className="nav-item">
                            <button className={isNewUser ? 'nav-link' : 'nav-link active'} onClick={() => setIsNewUser(false)}>Login</button>
                        </li>
                        <li className="nav-item">
                            <button className={isNewUser ? 'nav-link active' : 'nav-link'}onClick={() => setIsNewUser(true)}>Signup</button>
                        </li>

                    </ul>
                </div>
                <div className="card-body">
                <form onSubmit={handleSubmit}>
                    <div className="form-group">
                        <label htmlFor="username">Username</label>
                        <input
                        className="form-control" 
                        type="text" 
                        value={formData.username}
                        id="username" 
                        name="username" 
                        placeholder="Username" 
                        onChange={handleChange}/>
                    </div>
                    <div className="form-group">
                        <label htmlFor="password">Password</label>
                        <input 
                            className="form-control"
                            type="password" 
                            value={formData.password} 
                            id="password"
                            name="password" 
                            placeholder="Password" 
                            onChange={handleChange}/>
                    </div>
                    {isNewUser && 
                        <>
                        <div className="form-group">
                            <label htmlFor="first_name">First Name</label>
                            <input 
                                className="form-control"
                                type="text" 
                                value={formData.first_name} 
                                id="first_name"
                                name="first_name" 
                                placeholder="e.g. John" 
                                onChange={handleChange}/>
                        </div>
                        <div className="form-group">
                            <label htmlFor="last_name">Last Name</label>
                            <input 
                                className="form-control"
                                type="text" 
                                value={formData.last_name} 
                                id="last_name"
                                name="last_name" 
                                placeholder="e.g. Smith" 
                                onChange={handleChange}/>
                        </div>
                        <div className="form-group">
                            <label htmlFor="email">Email Address</label>
                            <input 
                                className="form-control"
                                type="text" 
                                value={formData.email} 
                                id="email"
                                name="email" 
                                placeholder="e.g. johnsmith@gmail.com" 
                                onChange={handleChange}/>
                        </div>
                        </>
                    }
                    <button className="btn btn-info">{isNewUser ? 'Create Account' : 'Log in'}</button>
                </form>
                    {formData.errors.length > 0 ? <Alert msgs={formData.errors}/> : null }
                </div>
            </div>

        </div>
    )
}

export default Login;

