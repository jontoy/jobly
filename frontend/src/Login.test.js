import React from 'react';
import { render } from '@testing-library/react';
import Login from './Login';
import { MemoryRouter } from 'react-router-dom';
import {UserProvider} from './testUtils';

it('renders without crashing', () => {
    render(
    <MemoryRouter>
        <UserProvider>
            <Login />
        </UserProvider>
    </MemoryRouter>
    );
});
  
it('matches snapshot when user is logged in', () => {
      const {asFragment} = render(
        <MemoryRouter>
            <UserProvider>
                <Login />
            </UserProvider>
        </MemoryRouter>
        );
    expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot when user is logged out', () => {
    const {asFragment} = render(
      <MemoryRouter>
          <UserProvider currentUser={null}>
              <Login />
          </UserProvider>
      </MemoryRouter>
      );
  expect(asFragment()).toMatchSnapshot();
});