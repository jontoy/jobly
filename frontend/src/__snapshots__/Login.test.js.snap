// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when user is logged in 1`] = `
<DocumentFragment>
  <div
    class="Login container"
  >
    <div
      class="card"
    >
      <div
        class="card-header"
      >
        <ul
          class="nav nav-tabs card-header-tabs ml-auto"
        >
          <li
            class="nav-item"
          >
            <button
              class="nav-link active"
            >
              Login
            </button>
          </li>
          <li
            class="nav-item"
          >
            <button
              class="nav-link"
            >
              Signup
            </button>
          </li>
        </ul>
      </div>
      <div
        class="card-body"
      >
        <form>
          <div
            class="form-group"
          >
            <label
              for="username"
            >
              Username
            </label>
            <input
              class="form-control"
              id="username"
              name="username"
              placeholder="Username"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="password"
            >
              Password
            </label>
            <input
              class="form-control"
              id="password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-info"
          >
            Log in
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when user is logged out 1`] = `
<DocumentFragment>
  <div
    class="Login container"
  >
    <div
      class="card"
    >
      <div
        class="card-header"
      >
        <ul
          class="nav nav-tabs card-header-tabs ml-auto"
        >
          <li
            class="nav-item"
          >
            <button
              class="nav-link active"
            >
              Login
            </button>
          </li>
          <li
            class="nav-item"
          >
            <button
              class="nav-link"
            >
              Signup
            </button>
          </li>
        </ul>
      </div>
      <div
        class="card-body"
      >
        <form>
          <div
            class="form-group"
          >
            <label
              for="username"
            >
              Username
            </label>
            <input
              class="form-control"
              id="username"
              name="username"
              placeholder="Username"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="password"
            >
              Password
            </label>
            <input
              class="form-control"
              id="password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-info"
          >
            Log in
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
